src/ngraph/pass/dyn_elimination.cpp

```diff
@@ -79,6 +82,50 @@ void pass::DynElimination::construct_transpose()
     add_matcher(transpose_matcher, transpose_callback, all_pass_property_off);
 }
 
+void pass::DynElimination::construct_broadcast()
+{
+    auto data_arg_label = make_shared<pattern::op::Label>(element::f32, Shape{1, 2, 3});
+    auto shape_arg_label =
+        make_shared<pattern::op::Label>(element::i64, Shape{3}, pattern::has_class<op::Constant>());
+    auto axes_arg_label =
+        make_shared<pattern::op::Label>(element::i64, Shape{0}, pattern::has_class<op::Constant>());
+
+    auto dyn_broadcast =
+        make_shared<op::DynBroadcast>(data_arg_label, shape_arg_label, axes_arg_label);
+
+    auto dyn_broadcast_callback =
+        [data_arg_label, shape_arg_label, axes_arg_label](pattern::Matcher& m) {
+            auto pattern_map = m.get_pattern_map();
+
+            auto data_arg = pattern_map[data_arg_label];
+            auto shape_arg = static_pointer_cast<op::Constant>(pattern_map[shape_arg_label]);
+            auto axes_arg = static_pointer_cast<op::Constant>(pattern_map[axes_arg_label]);
+
+            // TODO(amprocte): These should be redundant if the graph is validated. Necessary?
+            if (shape_arg->get_element_type() != element::i64 ||
+                shape_arg->get_output_partial_shape(0).is_dynamic() ||
+                shape_arg->get_output_shape(0).size() != 1 ||
+                axes_arg->get_element_type() != element::i64 ||
+                axes_arg->get_output_partial_shape(0).is_dynamic() ||
+                axes_arg->get_output_shape(0).size() != 1)
+            {
+                return false;
```

Comment:
We should probably report an error here.

Reply:
Agreed, except in the `.is_dynamic()` case. Will go ahead and slip that in (should have time, we have several `High Priority` PRs ahead of it :))

Reply:
73c7c8a if you want to double check my work here.